functests/common.sh

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env bash
+#
+# This file is part of MinIO Direct CSI
+# Copyright (c) 2021 MinIO, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+# Enable tracing if set.
+[ -n "$BASH_XTRACEFD" ] && set -ex
+
+export LV_DEVICE=
+export LUKS_DEVICE=
+export DIRECT_CSI_CLIENT=
+export DIRECT_CSI_VERSION=
+
+# usage: create_loop <newfile> <size>
+function create_loop() {
+    truncate --size="$2" "$1"
+    sudo losetup --find "$1"
+    sudo losetup --noheadings --output NAME --associated "$1"
+}
+
+function setup_lvm() {
+    loopdev=$(create_loop testpv.img 1G)
+    sudo pvcreate "$loopdev"
+    vgname="testvg$RANDOM"
+    sudo vgcreate "$vgname" "$loopdev"
+    sudo lvcreate --name=testlv --extents=100%FREE "$vgname"
+    LV_DEVICE=$(readlink -f "/dev/$vgname/testlv")
+}
+
+function setup_luks() {
+    loopdev=$(create_loop testluks.img 1G)
+    echo "mylukspassword" > lukspassfile
+    yes YES | sudo cryptsetup luksFormat "$loopdev" lukspassfile
+    sudo cryptsetup -v luksOpen "$loopdev" myluks --key-file=lukspassfile
+    LUKS_DEVICE=$(readlink -f /dev/mapper/myluks)
+}
+
+function install_directcsi() {
+    "${DIRECT_CSI_CLIENT}" install --image "direct-csi:${DIRECT_CSI_VERSION}"
+
+    pending=7
+    while [[ $pending -gt 0 ]]; do
+        echo "$ME: waiting for direct-csi pods to come up... ${pending}"
+        sleep ${pending}
+        count=$(kubectl get pods --field-selector=status.phase=Running --no-headers --namespace=direct-csi-min-io | wc -l)
+        pending=$(( pending - count ))
+    done
+
+    while true; do
+        echo "$ME: waiting for direct-csi to come up..."
+        sleep 5
+        if "${DIRECT_CSI_CLIENT}" info; then
+            return 0
+        fi
+    done
+}
+
+function _uninstall_directcsi() {
+    "${DIRECT_CSI_CLIENT}" uninstall "$@"
+
+    pending=7
+    while [[ $pending -gt 0 ]]; do
+        echo "$ME: waiting for direct-csi pods to go down... ${pending}"
```

Comment:
we wont delete the conversion webhook pods while uninstalling with v1.3.6. We implemented that change in v1.4.0 where we bring down conversion webhook deployment during uninstallation. 

(cancelling the CI workflow as the script waits for this condition to be satisfied)